Meadowlark-js/packages/meadowlark-core/src/metaed/MetaEdValidation.ts

```diff
@@ -77,16 +77,57 @@ export function validateQueryParametersAgainstSchema(
 ): string[] {
   // TODO: RND-67 removed the Joi-specific code to report invalid query parameters
   // RND-307 will restore as a part of understanding Ajv error metadata
+
+  let errors: string[] = [];
+
   const schema = {
     ...metaEdModel.data.meadowlark.jsonSchema,
     // Need to relax the validation such that no fields are "required"
     required: [],
   };
 
+  const parameterKeys = Object.keys(queryParameters);
+
+  /**
+   * Number and boolean are come through as strings, which causes ajv to fail validation of parameters
+   * This loops through all query parameters and checks the type against the schema, if the type is numeric or boolean
+   * it will attempt convert the value. If the conversion is good, we set the value back and ajv will validate the schema
+   * If the value can't be converted (i.e. a word is provided for a numeric value, we leave it and let ajv invalidate
+   * the schema).
+   */
+  // eslint-disable-next-line no-restricted-syntax
+  for (const keyValue of parameterKeys) {
```

Comment:
Why not follow eslint's advice here? I don't remember the exact reason, but it is my understanding that the following is a safer approach (perhaps because of the possibility of `parameterKeys` being modified during enumeration):

```suggestion
  Object.keys(queryParameters).forEach((key) => {
```

Reply:
Agree. We do occasionally override eslint on loops when the forEach function gets tangled up with async/await issues, but that's because using forEach would actually be wrong.

Reply:
was from the data store code where we were making async calls during the foreach, which is bad. I took that a little too much to heart. No async calls in this one, so fixed.